### ISSUE TITLE
Build triggered on MR approval even if option is disabled

### ISSUE THREAD

**Issue by chinhodado** [jenkinsci/gitlab-plugin#796]
## Issue

I upgraded the plugin to 1.5.8 and this option is broken. I have this option unchecked, but the build is still triggered upon an approval.

This option was broken a long time ago, then it was fixed, then broken again in the last few updates (I think either 1.5.7 or 1.5.8)

### Context
- **Gitlab plugin version**: 1.5.8
- **Gitlab version**: 11.0.0
- **Jenkins version**: 2.130
- **Job type**: Freestyle

**Comment by omehegan** [comment]
Duplicate of #734.

**Comment by chinhodado** [comment]
@omehegan I'm not sure #734 is the same as this one, instead it looks like it's the **opposite** to me

#734 is "Can't trigger builds on Accepted, Approved & Closed Merge Requests"

Whereas my issue is "Build on Approval still triggered even though the option is **disabled**"

**Comment by omehegan** [comment]
Sorry for misunderstanding your issue! What version of the plugin were you running before? You should be able to see this in the Jenkins plugin UI, where it shows the option to downgrade.

**Comment by chinhodado** [comment]
I'm pretty sure the regression was introduced in either 1.5.7 or 1.5.8. I've been using 1.5.6 for a long time without issue, and I've noticed the problem very recently, in the last week or so.

**Comment by omehegan** [comment]
I see, in 1.5.6 trigger on MR approval was broken completely, so this didn't affect you. But the change to fix it, which was just released, seems to not respect the unchecked option. @keijohyttinen does this make sense?

**Comment by keijohyttinen** [comment]
Hmm, haven't tested yet with newest official release 1.5.8 but I did test fix (https://github.com/jenkinsci/gitlab-plugin/pull/762) on top of release 1.5.6. With that self made package, our pipelines work fine with plugin options and we have even dedicated&separated pipelines for open MRs and accepted MRs. 
However, note that we have jenkins pipeline, not freestyle job type. I don't know what's difference of freestyle/pipeline job from plugin's standpoint. Could freestyle job cause the issue?

anyway, @chinhodado based on pipeline test results, suspect there might be something else which causes your problem.
What are your settings for the gitlab plugin?
Which gitlab hooks attached to job (push, merge requests, etc)?

I think, e.g. "_Rebuild open Merge Requests_" option set to "_On push to source or target branch_" overrides other settings and will trigger build on target branch push (equals to accepted MRs trigger).

**Comment by chinhodado** [comment]
This is my current configuration:

![image](https://user-images.githubusercontent.com/1388219/42452902-42c682bc-8359-11e8-8bdc-c78b8dd92708.png)

Build is only triggered on open merge request, and on push to source branch

**Comment by keijohyttinen** [comment]
@omehegan @chinhodado I did little investigation and opened MR https://github.com/jenkinsci/gitlab-plugin/pull/798 for your review. 
It's highly possible that "_Rebuild open Merge Requests_" option (i.e. State "_updated_") conflicts with "_approved_" action received from Gitlab EE. Moreover that might be the original reason to change triggering check in the first place, which eventually broke triggering of other states, and led me to revert it in https://github.com/jenkinsci/gitlab-plugin/pull/762.

I'm afraid but I've no chance to verify my possible fix in real life environment and to see what states/actions Gitlab EE really sends to Jenkins in case of MR approved.

I did extended tests to covers more possible use cases and if my assumption is correct, the change won't fail other triggers but will fix "approved" option for Gitlab EE.

**Comment by dskrvk** [comment]
We also noticed this regression in v.1.5.8 or 1.5.7. The job trigger settings are similar to OP's (rebuild on source push, build on approve disabled; triggering events are push, MR opened, comments (with a regex filter)).

**Comment by eddie3716** [comment]
This has been a problem for me and a lot of other people since the change #664 was introduced back in December 2017.  The author of the change in #664 doesn't even use the plugin in a typical manner, 
 and so failed ensure that no regressive changes were introduced.  Previous attempts to shore up the plugin from this broken state (#711) have been inadequate half-measures.  Some one is yet again reporting this problem.  I got around this by keeping a version of the 1.5.0 plugin available, but a recent on-prem Gitlab upgrade is no longer compatible with the 1.5.0 plugin, so I'm forced to use the latest version (1.5.9) to have any semblance of a working integration.

**Comment by omehegan** [comment]
@eddie3716 I wasn't aware of any issues after #664 was merged, except for #711 which corrected the config persistence issue. That was released in version 1.5.5. Did you open another issue about this previously?

**Comment by omehegan** [comment]
@dskrvk @eddie3716 please see the snapshot release in #798 and let me know if this fixes the issue for you.

**Comment by omehegan** [comment]
This was released in version 1.5.9 of the plugin.

**Comment by dskrvk** [comment]
The snapshot build fixed it for me.

**Comment by eddie3716** [comment]
1.5.9 fixes the issues for me.  Thank you!

**Comment by nbauma109** [comment]
> 1.5.9 fixes the issues for me. Thank you!

Very sorry to resurrect this old issue but I'm precisely using 1.5.9 and have the issue.
 - On the Gitlab webhook part, Merge request events is ticked.
- On Jenkins job, I made sure Opened Merge Request Events is ticked and Approved Merge Requests (EE-only) is unticked.
